go.mod

```diff
@@ -361,3 +362,5 @@ replace github.com/go-openapi/spec v0.20.5 => github.com/go-openapi/spec v0.20.6
 // Also see the source version from the datadog agent
 // https://github.com/DataDog/datadog-agent/blob/main/go.mod#L26
 replace github.com/theupdateframework/go-tuf v0.3.1 => github.com/DataDog/go-tuf v0.3.0--fix-localmeta
+
+replace github.com/DataDog/agent-payload/v5 v5.0.59 => github.com/DataDog/agent-payload/v5 v5.0.49 // indirect
```

Comment:
Why?  This needs a comment explaining why the replacement is made and ideally when it is no longer needed.  Is there any reason that this particular version needs to be used and we're not just excluding a bad version?

Reply:
Probably, we could also add link : https://pkg.go.dev/github.com/DataDog/agent-payload/v5?tab=versions and desc like [v5.0.59](https://pkg.go.dev/github.com/DataDog/agent-payload/v5@v5.0.59) is not a right version.
Wondering if we have to alert otel community as well about this.

Reply:
Thanks, added comment. 
reading github.com/DataDog/agent-payload/go.mod at v5.0.59 is throwing error `unknown revision v5.0.59`. 
This seems like some versioning issue as this release is not part of github repo: https://github.com/DataDog/agent-payload/releases, but is published at https://pkg.go.dev/github.com/DataDog/agent-payload/v5?tab=versions, although the version numbering and dates seems not inline and are very recent. So, selected the last available version published 3 weeks ago.

Reply:
The upstream `go.mod` file [doesn't appear to have this issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/76771863a86bd4d8aecb5a8446d23b26a92b36cc/go.mod#L222). Could this be caused by `GOPROXY=DIRECT` on our local machines?  Should we try running running the build on GitHub Actions without this to see if it works?

Reply:
I have no issues in my local too but when i tried its updating my go.mod as we see in upstream, wouldn't be sure if upstream didn't have the issue. tried this in my local which say its found but has no package in it.
<img width="1377" alt="Screen Shot 2023-01-04 at 9 13 38 AM" src="https://user-images.githubusercontent.com/41936996/210611760-82bc5037-a5c5-4dbc-98de-029091d56853.png">

Reply:
> I have no issues in my local too and go.mod is updates as we see in upstream. wouldn't be sure if upstream didn't have the issue, tried this in my local which say its found but has no package in it.
> <img alt="Screen Shot 2023-01-04 at 9 13 38 AM" width="1377" src="https://user-images.githubusercontent.com/41936996/210611760-82bc5037-a5c5-4dbc-98de-029091d56853.png">

disregard

Reply:
I have updated PR (gomod and gosum) with DataDog/agent-payload/v5 v5.0.59. Build is passing on GitHub and not on local. Yes, `GOPROXY` is set to `direct` on local. but not sure why build is passing on GitHub env and not on local?

Reply:
Excluded v5.0.59 as discussed.